app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
                 }
 
                 bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
+                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
+
+                bgReading.save();
+                bgReading.perform_calculations();
+                Notifications.getInstance(context).notificationSetter(context);
+                BgSendQueue.addToQueue(bgReading, "create", context);
+            }
+        }
+        Log.w("BG GSON: ",bgReading.toS());
+
+        return bgReading;
+    }
+    public static BgReading create(double raw_data, Context context, Long timestamp) {
```

Comment:
Dexbridge gets filtered_value. That's why the new function. To make sure it is set, rather than just being set to raw_data as per the existing function. It can probably be done more efficiently, by create(raw_data, context, timestamp calling this new function with create(raw_data, raw_data, context, timestamp). But otherwise the filtered_data function needs to be used by dexbridge code. 

 
John Stevens
M - +61 418526349

> On 20 May 2015, at 07:31, tzachi-dar <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java:
> 
> > @@ -323,6 +326,70 @@ public static BgReading create(double raw_data, double filtered_data, Context co
> >                  }
> >  
> >                  bgReading.calculated_value = Math.min(400, Math.max(40, bgReading.calculated_value));
> > +                Log.w(TAG, "create: NEW VALUE CALCULATED AT: " + bgReading.calculated_value);
> > +
> > +                bgReading.save();
> > +                bgReading.perform_calculations();
> > +                Notifications.getInstance(context).notificationSetter(context);
> > +                BgSendQueue.addToQueue(bgReading, "create", context);
> > +            }
> > +        }
> > +        Log.w("BG GSON: ",bgReading.toS());
> > +
> > +        return bgReading;
> > +    }
> > +    public static BgReading create(double raw_data, Context context, Long timestamp) {
> Please note that alerts code looks at filtered data as well. If you can't get the real value, need to set there the rawvalue. Thanks Tzachi
> …
> —
> Reply to this email directly or view it on GitHub.
>